### PR TITLE
[DT-10][risk=no] Consolidate API calls that have many arguments

### PR DESCRIPTION
**Pull request by chenchals**
Description:
---
- Consolidated API calls that have many arguments. Merged the 2-calls below
  - `CohortBuilderController.findCriteriaAutoComplete`
  - `CohortBuilderController.findSurveyAutoComplete`
- Updated Swagger definition
- Updated API calls, 
- Updated UI calls 
- Updated API tests
- there were no explicit UI tests for these calls(?)
- Deployed app to local and verified autocomplete is working as expected

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [x] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

**Review by dolbeew — APPROVED** [review]
UI LGTM

File: api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java [hunk 1]
```diff
@@ -61,34 +60,22 @@ public class CohortBuilderController implements CohortBuilderApiDelegate {
...
+    if (Domain.SURVEY.equals(Domain.fromValue(criteriaSearchRequest.getDomain()))) {
+      validateSurveyName(criteriaSearchRequest.getSurveyName());
+      return ResponseEntity.ok(
+          new CriteriaListResponse()
+              .items(cohortBuilderService.findCriteriaAutoComplete(criteriaSearchRequest)));
+    } else {
+      validateType(criteriaSearchRequest.getType());
+      return ResponseEntity.ok(
+          new CriteriaListResponse()
+              .items(cohortBuilderService.findCriteriaAutoComplete(criteriaSearchRequest)));
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java:77]: Common part can be extracted from if statement
```suggestion
    if (Domain.SURVEY.equals(Domain.fromValue(criteriaSearchRequest.getDomain()))) {
      validateSurveyName(criteriaSearchRequest.getSurveyName());
    } else {
      validateType(criteriaSearchRequest.getType());
    }
    return ResponseEntity.ok(
        new CriteriaListResponse()
            .items(cohortBuilderService.findCriteriaAutoComplete(criteriaSearchRequest)));
```

**Review by freemabd — COMMENTED** [review]
When searching auto complete for Surveys i get a CORS error and infinite spinner
Screen Shot 2022-10-14 at 9 44 35 AM

When searching auto complete for other domains i get a 500 error
Screen Shot 2022-10-14 at 9 46 21 AM

File: api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java (hunk 1 above)
Comment (chenchals) [reply api/src/main/java/org/pmiops/workbench/api/CohortBuilderController.java:77]: aah -- one more step...missed done

**Comment by chenchals** [comment]
> When searching auto complete for Surveys i get a CORS error and infinite spinner Screen Shot 2022-10-14 at 9 44 35 AM
> 
> When searching auto complete for other domains i get a 500 error Screen Shot 2022-10-14 at 9 46 21 AM

I do not get the infinite spinner., Can you recheck your local?
image

All Surveys:
image

Or other domains I checked

**Review by freemabd — APPROVED** [review]
Thanks

**Pull request merged**